### DISCUSSION TITLE
Feature request: Downloading a repository should also download all the submodules

### DISCUSSION THREAD

**Discussion by simonbs** [community/community#20797] [Other Feature Feedback, Questions, & Ideas]
Hi 👋

I've got a feature request and I couldn't find a better place to post it than here. Please let me know if this isn't the proper place to post it. Here's my feature request.

**Downloading a repository should also download all the submodules**

When users download a repository by selecting Code -> Download ZIP, the downloaded zip should include clones of all the submodules in the repository, and naturally, this would have to be recursive. The below image shows a screenshot of the Download ZIP button that I'm referring to.

<img width="938" alt="Screenshot 2022-07-18 at 12 28 28" src="https://user-images.githubusercontent.com/830995/179493383-c2909c9d-1dc9-48da-b5be-470fd48f2b03.png">

Today users can download a repository but be unable to build the code in the repository because it doesn't in fact contain the entire codebase. The downloaded zip file doesn't include the Git files, and as such, users can't manually clone the submodules either. Users are left with no option but to clone the repository, making the Download ZIP less useful for repositories that contain submodules.

**Comment by Kilozen** [comment]
I posted a similar request, except that mine refers to the .zips automatically created in the default **'Release' process**: https://github.com/community/community/discussions/31795 

(I assume you know that a user could **Clone** your whole project using the --recurse-submodules switch and it would pick up all the submodules.  But it seems you are referring to a simpler **Download Zip** of any code branch, and not a clone of the whole git repository.  So **I agree** with your request.)

**Comment by GromNaN** [comment]
We are going to use GitHub Archives to download and install PHP extensions, and having the submodules includes would be very much appreciated. https://github.com/php/pie/issues/39
